File: Java/plusOne.java
package Leetcodes;

import java.util.ArrayList;
import java.util.List;

public class plusOne {
    public static int[] plusOne(int[] digits){
       for(int i = digits.length - 1; i>=0; i--){
           if (digits[i] < 9){
               digits[i]++;
               return digits;
           }
           digits[i] = 0; //carry over
       }
       // if all digits are nine then we need an extra digit at the front
        int[] result = new int[digits.length + 1];
        result[0] = 1;
        return result;
    }

    public static void main(String[] args) {
        int[] digits = {1,2,3};
        int[] answer = plusOne(digits);
        System.out.print(answer);
    }
}

File: Python/plusOne.py
def plusOne(digits : list[int]) -> list[int]:
  '''
  s = [str(num) for num in digits]
  s = "".join(s)
  ans = str(int(s) + 1)

  answer = [int(digit) for digit in ans]
  return answer
  '''

  for i in range(len(digits)-1, -1, -1):
    if digits[i] < 9:
      digits[i] += 1
      return digits
    digits[i] = 0 #carry over

    # assuming all are 9's
    return [1] + [0] * len(digits)

  

digits = [1,2,3]
# Output : 123 + 1 = 124 -> [1,2,4]
answer = plusOne(digits)
print(answer)